daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SValue.scala

```diff
@@ -34,11 +33,8 @@ sealed trait SValue {
       case SUnit => V.ValueUnit
       case SDate(x) => V.ValueDate(x)
       case SStruct(fields, svalues) =>
-        V.ValueStruct(
-          Struct.assertFromSeq(
-            (fields.iterator zip svalues.iterator().asScala.map(_.toValue)).toSeq
-          ),
-        )
+        val valuesIterator = svalues.iterator().asScala.map(_.toValue)
+        V.ValueStruct(fields.mapValues(_ => valuesIterator.next()))
```

Comment:
With my limited Scala knowledged, I would assume the code below avoids the adaptor for `map` on the iterator. Does it?
```suggestion
        val valuesIterator = svalues.iterator().asScala
        V.ValueStruct(fields.mapValues(_ => valuesIterator.next().toValue))
```